src/OmniSharp.Roslyn.CSharp/Services/Refactoring/V2/CodeFixesForProjects.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using System.Reflection;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using OmniSharp.Services;
+using OmniSharp.Utilities;
+
+namespace OmniSharp.Roslyn.CSharp.Services.Refactoring.V2
+{
+    [Shared]
+    [Export(typeof(CodeFixesForProjects))]
+    public class CodeFixesForProjects
+    {
+        private readonly ConcurrentDictionary<string, IEnumerable<CodeFixProvider>> _codeFixCache = new ConcurrentDictionary<string, IEnumerable<CodeFixProvider>>();
```

Comment:
using IEnumerable in the state seems dangerous. if it is guaranteed to not change, then, just pay one time cost to ImmutableArray or something and be safe.